### PR TITLE
Bug/5106 - Fix AdSense V2 main logic

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #5106 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/adsense/components/setup/v2/SetupAccount.js [hunk 1]
```diff
@@ -70,23 +70,29 @@ export default function SetupAccount( { account } ) {
...
 	);
 
-	const afcClientID = determineClientID( { clients: clients || [] } );
+	const { setClientID, setAccountStatus, setSiteStatus } = useDispatch(
+		MODULES_ADSENSE
+	);
 
-	const { setClientID, setAccountStatus } = useDispatch( MODULES_ADSENSE );
+	useEffect( () => {
+		if ( afcClient?._id ) {
```
Comment (techanvil) [inline assets/js/modules/adsense/components/setup/v2/SetupAccount.js:78]: This condition also needs to check that `clientID !== afcClient._id`
```suggestion
		if ( afcClient?._id && clientID !== afcClient._id ) {
```

File: assets/js/modules/adsense/components/setup/v2/SetupMain.js [hunk 2]
```diff
@@ -180,25 +131,15 @@ export default function SetupMain() {
...
-			return;
-		}
-
-		if (
-			( accounts?.length === 1 && ! previousAccountID ) ||
-			( accounts?.length === 1 &&
-				accounts.findIndex( ( { _id } ) => _id === accountID ) === -1 )
+			accounts?.length === 1 &&
+			( ! accountID ||
+				accounts.findIndex( ( { _id } ) => _id !== accountID ) ) === 0
```
Comment (techanvil) [inline assets/js/modules/adsense/components/setup/v2/SetupMain.js:136]: This is an overly complex condition and should be simplified (we know there's only 1 account as the test for `accounts?.length === 1` has already returned `true`):

```suggestion
				accounts[ 0 ]._id !== accountID
```

**Review by techanvil — CHANGES_REQUESTED** [review]
Looks good @hussain-t, just a couple of minor comments from me.

File: assets/js/modules/adsense/components/setup/v2/SetupMain.js (hunk 2 above)
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/v2/SetupMain.js:136]: Thanks @techanvil, that's a good catch 👍

**Review by techanvil — APPROVED** [review]
LGTM, nice one @hussain-t!

**Pull request merged**